zio-http/src/main/scala/zio/http/api/Middleware.scala

```diff
@@ -104,6 +105,21 @@ object Middleware {
   def addCookieZIO[R](cookie: ZIO[R, Nothing, Cookie[Response]]): Middleware[R, Unit, Cookie[Response]] =
     fromFunctionZIO(MiddlewareSpec.addCookie)(_ => cookie)
 
+  def withAccept(value: CharSequence): Middleware[Any, Unit, Accept] =
```

Comment:
The thing I don't like about these middlewares is that they produce constant (static) headers but they are typed with non-Unit values, which suggests (to me) that they are more dynamic.

If indeed these will only ever produce the same static header, you can do `mapOut(_.unit(/* static value*/))` in order to turn the type parameter from `Accept` into `Unit`. You will have to provide a static `Accept` value, but it looks like you have that so won't be a problem.

Reply:
@jdegoes I've thought about this several times since Friday and to be honest, I can see it either way. Ultimately, i'm happy to defer to your opinion here. Are you asking me to change the return type to `Unit`? or are you okay with this PR as-is?